components/frontend/src/utils.js

```diff
@@ -237,14 +237,9 @@ export function visibleMetrics(metrics, metricsToHide, hiddenTags) {
     if (metricsToHide === "all") {
         return {}
     }
-    const visible = {}
-    Object.entries(metrics).forEach(([metric_uuid, metric]) => {
-        if (hideMetric(metric, metricsToHide, hiddenTags)) {
-            return
-        }
-        visible[metric_uuid] = metric
-    })
-    return visible
+    return Object.fromEntries(
+        Object.entries(metrics).filter(([_, metric]) => !hideMetric(metric, metricsToHide, hiddenTags)),
+    )
```

Comment:
Refactor by using `Object.fromentries` instead of manually constructing the filtered object.